rest_framework/templates/rest_framework/base.html

```diff
@@ -22,7 +22,9 @@
 
         <link rel="stylesheet" type="text/css" href="{% static "rest_framework/css/prettify.css" %}"/>
         <link rel="stylesheet" type="text/css" href="{% static "rest_framework/css/default.css" %}"/>
-        {% if code_style %}<style>{{ code_style }}</style>{% endif %}
+        {% if code_style %}
+          <style{% if request.csp_nonce %} nonce="{{request.csp_nonce}}"{% endif %}>{{ code_style }}</style>
```

Comment:
django-csp adds this `csp_nonce` in middleware. I'm not sure if including this in DRF is a good idea.

Reply:
if that is necessary then we could consider adding it

Reply:
I think above is fine as-is. If users install django-csp and enable nonces, the nonce will get automatically added to the <style> element:
```
{% if code_style %}
          <style nonce="{{request.csp_nonce}}">{{ code_style }}</style>
{% endif %}
```

If users don't have django-csp installed or don't use nonces in their CSP, above will quietly fall back to the old behavior:
```
{% if code_style %}
          <style>{{ code_style }}</style>
{% endif %}
```

I'll test this locally in a bit, then approve if I see no issues. If there's no further feedback here I'll probably merge tomorrow morning.

Reply:
Actually it looks like this causes issues. If you follow the django-csp docs and add its provided CSPMiddleware, accessing this ```request.csp_nonce``` property in any template causes the nonce value to be automatically included in your CSP. This has some side-effects...

According to the django-csp docs, you can control what directives use the nonce through the ```CSP_INCLUDE_NONCE_IN``` setting. When you don't set ```CSP_INCLUDE_NONCE_IN```, or set it to an empty tuple / list, the nonce ends up in the ```default-src``` directive. This conflicts with a policy like ```default-src: 'none'```, because it gets automatically converted to ```default-src: 'none' 'nonce-1234'```. The 'none' value is then ignored because a nonce value is present.

It's technically possible to work around this by setting ```CSP_INCLUDE_NONCE_IN``` to a particular directive. Then the nonce value is only included in the directives given, and isn't included in the 'default-src directive anymore. But this will cause issues with policies like ```script-src: 'unsafe-inline'```, which will get automatically converted to ```script-src: 'unsafe-inline' 'nonce-1234'```. The 'unsafe-inline' value is then ignored because a nonce value is present.

So including the nonce here essentially forces people to use nonces, even if they don't want to. I don't see any good workaround for this. Removing the nonce just breaks the inline style, and the styling is dynamically generated so moving it to a static file is tricky. I may push a toy commit with an ugly hack people can review. Thoughts welcome.

Reply:
This broken styling is (I think) only related to syntax highlighting using CSS that's dynamically generated by Pygments.

You won't see this error if Pygments is not installed, or if your CSP contains 'unsafe-inline'. You can also ignore this error if you don't care about syntax highlighting.

If you want an error-free strong CSP with syntax highlighting enabled:

1. Make sure you have a strong CSP like ```style-src: 'self'```
2. Visit your page in Google Chrome
3. In the Developer Tools console, note the error message about CSP and the hash
4. Add that hash to your CSP, like ```style-src: 'self' 'sha256-1234'```